resources/type-system/flow-analysis.md

```diff
@@ -683,7 +705,15 @@ TODO: Add missing expressions, handle cascades and left-hand sides accurately
   (E) S` then:
   - Let `before(E) = conservativeJoin(before(N), assignedIn(N), capturedIn(N))`.
   - Let `before(S) = split(true(E))`.
-  - Let `after(N) = join(false(E), unsplit(break(S))`
+  - Let `after(N) = inheritTested(join(false(E), unsplit(break(S))), after(S))`.
+
+- **for statement**: If `N` is a for statement of the form `for (D; C; U) S`,
+  then:
+  - Let `before(D) = before(N)`.
+  - Let `before(E1) = conservativeJoin(after(D), assignedIn(N), capturedIn(N))`.
+  - Let `before(S) = split(true(C))`.
+  - Let `before(U) = merge(after(S), continue(S))`.
+  - Let `after(N) = inheritTested(join(false(E), unsplit(break(S))), after(U))`.
```

Comment:
Wouldn't `E` be `C`?

Also, can we now add a new bullet point along the lines of this?:

```
- **for-in statement**: If `N` is a for-in statement of the form
  `for (D in I) S` then the flow analysis is performed according
  to the desugaring of `N`. *The desugaring is specified in the
  language specification, and it yields a `while` loop.*
```

Reply:
Fixed!